toplevel/coqargs.ml

```diff
@@ -464,12 +464,8 @@ let parse_args ~help ~init arglist : t * string list =
 
     |"-w" | "-W" ->
       let w = next () in
-      if w = "none" then
-        (CWarnings.set_flags w; oval)
-      else
-        let w = CWarnings.get_flags () ^ "," ^ w in
-        CWarnings.set_flags (CWarnings.normalize_flags_string w);
-        oval
+      if w = "none" then add_set_option oval ["Warnings"] (Stm.OptionSet(Some w))
```

Comment:
Why this special case?

Reply:
It was already there. IIRC the code in cWarning checks if the *entire* string is "none" and in that case it resets.
Maybe I still don't get exactly how the goption thing works.

Reply:
>It was already there

Right, I read too fast. It looks like premature optimisation but it's fine I guess.